lib/redis/client.rb

```diff
@@ -247,14 +247,30 @@ def read
       io do
         value = connection.read
         @pending_reads -= 1
+
+        bytes = 0
+        if !value.nil?
+          bytes += value.size
+        end
+
+        ActiveSupport::Notifications.instrument('redis-rb.read.bytes', extra: bytes) do
```

Comment:
ActiveSupport is not a dependency of this gem. If anyone uses this gem in an application that doesn't depend on ActiveSupport already, this code will break.